tests/server/api/test_server.py

```diff
@@ -405,3 +412,62 @@ async def test_changing_database_breaks_cache(self, enable_memoization):
             await _memoize_block_auto_registration(test_func)()
 
         assert test_func.call_count == 2
+
+
+class TestSubprocessASGIServer:
```

Comment:
would it be worth adding a test that checks the case where we call `start()` more than once in a row for some reason?

Reply:
Added in https://github.com/PrefectHQ/prefect/pull/14722/commits/77862d8cb13052e3fafefc966317e1d9feccb22b!